src/Utilities/Registration.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Utilities/TMPL.hpp"
+
+/// \ingroup UtilitiesGroup
+/// \brief Helpers for derived class registration
+///
+/// SpECTRE's factory mechanism requires that the base class contain a
```

Comment:
maybe say here "a list of all derived classes (in the special nested type alias `creatable_classes`)" or something along those lines so that it's clear for people where exactly this info is kept.